geth/node/rpc.go

```diff
@@ -94,6 +107,65 @@ func (c *RPCManager) Call(inputJSON string) string {
 	return c.makeJSONErrorResponse(ErrRPCServerTimeout)
 }
 
+// callNodeStream delivers giving request and body content to the external ethereum
+// (infura) RPCServer to process the request and returns response.
+func (c *RPCManager) callUpstreamStream(config *params.NodeConfig, body io.Reader) ([]byte, error) {
+	upstreamURL, err := url.Parse(config.UpstreamConfig.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq, err := http.NewRequest("POST", upstreamURL.String(), body)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := http.Client{
+		Timeout: 20 * time.Second,
+	}
+
+	res, err := httpClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+
+	if respStatusCode := res.StatusCode; respStatusCode != http.StatusOK {
+		log.Error("handler returned wrong status code", "got", respStatusCode, "want", http.StatusOK)
+		return nil, ErrRPCServerCallFailed
+	}
+
+	defer res.Body.Close()
```

Comment:
Sorry, I haven't noticed that before. This call should be before checking status code. You can have a status code different from 200 and `res.Body` won't be closed.